addons/giltracker/giltracker.lua

```diff
@@ -0,0 +1,182 @@
+        (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'giltracker'
+_addon.author = 'sylandro'
+_addon.version = '1.0.0'
+_addon.language = 'English'
+
+config = require('config')
+images = require('images')
+texts = require('texts')
+
+hideKey = 70
+is_hidden_by_cutscene = false
+is_hidden_by_key = false
+
+defaults = {}
+defaults.HideKey = 70
+defaults.GilText = {}
+defaults.GilText.bg = {}
+defaults.GilText.bg.alpha = 100
+defaults.GilText.bg.red = 0
+defaults.GilText.bg.green = 0
+defaults.GilText.bg.blue = 0
+defaults.GilText.bg.visible = false
+defaults.GilText.text = {}
+defaults.GilText.text.font = 'sans-serif'
+defaults.GilText.text.fonts = {'Arial','Trebuchet MS'}
+defaults.GilText.text.size = 9
+defaults.GilText.flags = {}
+defaults.GilText.flags.italic = true
+defaults.GilText.flags.bold = false
+defaults.GilText.flags.right = true
+defaults.GilText.flags.bottom = true
+defaults.GilText.pos = {}
+defaults.GilText.pos.x = -285
+defaults.GilText.pos.y = -35
+defaults.GilText.text.alpha = 255
+defaults.GilText.text.red = 253
+defaults.GilText.text.green = 252
+defaults.GilText.text.blue = 250
+defaults.GilText.text.stroke = {}
+defaults.GilText.text.stroke.alpha = 200
+defaults.GilText.text.stroke.red = 50
+defaults.GilText.text.stroke.green = 50
+defaults.GilText.text.stroke.blue = 50
+defaults.GilText.text.stroke.width = 2
+defaults.GilText.text.visible = true
+defaults.GilImage = {}
+defaults.GilImage.color = {}
+defaults.GilImage.color.alpha = 255
+defaults.GilImage.color.red = 255
+defaults.GilImage.color.green = 255
+defaults.GilImage.color.blue = 255
+defaults.GilImage.visible = true
+
+local settings = config.load(defaults)
+config.save(settings)
+
+settings.GilImage.texture = {}
+settings.GilImage.texture.path = windower.addon_path..'gil.png'
+settings.GilImage.texture.fit = true
+settings.GilImage.size = {}
+settings.GilImage.size.height = 23
+settings.GilImage.size.width = 23
+settings.GilImage.draggable = false
+settings.GilImage.repeatable = {}
+settings.GilImage.repeatable.x = 1
+settings.GilImage.repeatable.y = 1
+
+gil_image = images.new(settings.GilImage)
+gil_text = texts.new(settings.GilText)
+
+windower.register_event('load',function()
+    if windower.ffxi.get_info().logged_in then
+        initialize()
+    end
+end)
+
+windower.register_event('login', function()
+    initialize()
+end)
+
+windower.register_event('logout', function(...)
+    hide()
+end)
+
+windower.register_event('add item', function(...)
+    update_gil()
+end)
+
+windower.register_event('remove item', function(original, modified, original_mode, modified_mode, blocked)
```

Comment:
The function signature is wrong, it looks like the one for the `incoming text` event. Not that Lua cares ¯\ (ツ) /¯

`remove item` and `add item` have the same signature: `function(bag, slot, id, count) --[[ ... ]] end`

Which is why it surprises me that the below code works the way you did it... are you sure it does. Again, because Lua stupidly doesn't error when calling string functions on numbers, this silently passes and returns `nil`. You would need to convert the item ID to an item using the resources first to compare the name:
```lua
windower.register_event('remove item', function(bag, slot, id, count)
    local item = res.items[id]
    if item.english:match('Gil') then -- it's also spelled with a capital G
        update_gil()
    end
end)
```

However, this would be true on items like `Gilt tapestry` as well, since it contains `gil`. You'd be best to either check for the whole name (item.english =='Gil'), or save yourself extra effort and just go by ID and save the entire resource lookup:
```lua
    if id == 0xFFFF then
        update_gil()
    end
```